dynawo/sources/Models/CPP/ModelLoadRestorativeWithLimits/DYNModelLoadRestorativeWithLimits.cpp

```diff
@@ -0,0 +1,577 @@
+    if (getIsInitProcess())
+      return;
+    if (!isConnected()) {
+      jt.changeCol();
+      jt.addTerm(UfRawYNum_ + rowOffset, 1);
+      jt.changeCol();
+      jt.addTerm(UfYNum_ + rowOffset, 1);
+      jt.changeCol();
+      jt.addTerm(IrYNum_ + rowOffset, 1);
+      jt.changeCol();
+      jt.addTerm(IiYNum_ + rowOffset, 1);
+      jt.changeCol();
+      jt.addTerm(PYNum_ + rowOffset, 1);
+      jt.changeCol();
+      jt.addTerm(QYNum_ + rowOffset, 1);
+    } else {
+      jt.changeCol();
+      jt.addTerm(UfRawYNum_ + rowOffset, 1.0 + cj * Tf_);
+      jt.addTerm(UrYNum_ + rowOffset, -Ur()/U());
+      jt.addTerm(UiYNum_ + rowOffset, -Ui()/U());
+      jt.changeCol();
+      jt.addTerm(UfYNum_ + rowOffset, 1);
+      if (UMaxPuReached_ == false && UMinPuReached_ == false) {
+        jt.addTerm(UfRawYNum_ + rowOffset, -1);
+      }
+      jt.changeCol();
+      jt.addTerm(IrYNum_ + rowOffset, 1);
+      jt.addTerm(PYNum_ + rowOffset, -Ur() / U2());
+      jt.addTerm(QYNum_ + rowOffset, -Ui() / U2());
+      jt.addTerm(UrYNum_ + rowOffset, - (P() * U2() - 2 * Ur() * (P() * Ur() + Q() * Ui())) / (U2() * U2()));
+      jt.addTerm(UiYNum_ + rowOffset, - (Q() * U2() - 2 * Ui() * (P() * Ur() + Q() * Ui())) / (U2() * U2()));
+      jt.changeCol();
+      jt.addTerm(IiYNum_ + rowOffset, 1);
+      jt.addTerm(PYNum_ + rowOffset, -Ui() / U2());
+      jt.addTerm(QYNum_ + rowOffset, Ur() / U2());
+      jt.addTerm(UiYNum_ + rowOffset, - (P() * U2() - 2 * Ui() * (P() * Ui() - Q() * Ur())) / (U2() * U2()));
+      jt.addTerm(UrYNum_ + rowOffset, (Q() *  U2() + 2 * Ur() * (P() * Ui() - Q() * Ur())) / (U2() * U2()));
+      jt.changeCol();
+      jt.addTerm(PYNum_ + rowOffset, 1);
+      jt.addTerm(UrYNum_ + rowOffset, - P_dUr(Ur(), Ui(), U(), U2()));
+      jt.addTerm(UiYNum_ + rowOffset, - P_dUi(Ur(), Ui(), U(), U2()));
+      jt.addTerm(UfYNum_ + rowOffset, - P_dUf(U()));
+      jt.changeCol();
+      jt.addTerm(QYNum_ + rowOffset, 1);
+      jt.addTerm(UrYNum_ + rowOffset, - Q_dUr(Ur(), Ui(), U(), U2()));
+      jt.addTerm(UiYNum_ + rowOffset, - Q_dUi(Ur(), Ui(), U(), U2()));
+      jt.addTerm(UfYNum_ + rowOffset, - Q_dUf(U()));
+    }
+  }
+
+  void
+  ModelLoadRestorativeWithLimits::evalG(const double& /*t*/) {
+    gLocal_[0] = (doubleNotEquals(UfRaw(), UMaxPu_) && UfRaw() > UMaxPu_) ? ROOT_UP : ROOT_DOWN;
+    gLocal_[1] = (doubleNotEquals(UfRaw(), UMinPu_) && UMinPu_ - UfRaw() > 0) ? ROOT_UP : ROOT_DOWN;
+  }
+
+  void
+  ModelLoadRestorativeWithLimits::setGequations() {
+    if (getIsInitProcess())
+      return;
+    if (isConnected()) {
+      if (gLocal_[0] == ROOT_UP) {
+        gEquationIndex_[0] = std::string("UMaxPu limit reached by Load: ").append(staticId());
+      }
+      if (gLocal_[1] == ROOT_UP) {
+        gEquationIndex_[1] = std::string("UMinPu limit reached by Load: ").append(staticId());
+      }
+    }
+  }
```

Comment:
In the implementation itself, no condition on isConnected(). It could be added there in evalG or removed here.